bindings/CXX11/adios2/cxx11/Engine.h

```diff
@@ -21,6 +21,32 @@
 namespace adios2
 {
 
+namespace detail
+{
+template <typename... Ts>
+struct void_t
+{
+};
+
+template <typename C, typename = void>
+struct is_container : std::false_type
```

Comment:
I propose having a verbose static_assert to inform the user that the type passed does not fulfill the traits "Contract

Reply:
You can't really have a `static_assert()` in there, since this is used for SFINAE, ie., you don't want an error, you just want the non-container argument to not be considered in this case and have it try the other possible overloads.